### PR TITLE
Release 0.3 - smart-notifier support & status endpoint

### PR DESCRIPTION
**Pull request by sbellity**

### CODE REVIEW

File: manifest.json [hunk 1]
```diff
@@ -62,18 +56,26 @@
...
     },
     {
-      "name": "excluded_domains_hero",
-      "type": "string",
-      "headerTemplate": "The above list of domains is a blacklist. You need to *explicitely specify one or more domains* for which enrichment shall be skipped.<br>If you do not specify any domain, the connector will perform a call to the Datanyze API for every user matching one of the whitelisted segments. Please note that each call will count towards your monthly rate limit.",
-      "format": "information",
-      "title": "Blacklisted Domains"
+      "name":"handle_accounts",
+      "title": "Handle Accounts",
+      "description":"If active, Data will be written at Account level in addition to User level. We will also use the Account domain field to query Datanyze - This overwrites the 'Company Domain' setting.",
```
Comment (michaloo) [inline manifest.json]: So we don't want to give similar setting to pick custom account trait to use as company domain?

File: server/lib/datanyze.js [hunk 2]
```diff
@@ -34,11 +35,15 @@ export default class DatanyzeClient {
       json: true,
       resolveWithFullResponse: true,
       qs: { token, email, ...params }
-    }).then((response) => {
+    }).then(response => {
       const body = response.body;
       if (body && response.statusCode === 200) {
+        if (path !== "limits") {
```
Comment (michaloo) [inline server/lib/datanyze.js]: `limits` call does not count towards the api rate limit?

File: server/lib/update-user.js [hunk 3]
```diff
@@ -1,141 +1,222 @@
...
-        return Promise.resolve();
-      }
+        if (!matchingSegments && !isBatch) {
+          return skip({ reason: "datanyze.user.segments_skip" });
+        }
 
-      asUser.logger.info("outgoing.user.start", { domain });
+        // Skip because can't find a Domain name to match
+        const rawDomain =
+          handle_accounts && account.domain
```
Comment (michaloo) [inline server/lib/update-user.js]: Can we have a situation that `account.domain` is empty? If so, and we have a enabled `handled_accounts` we would end up looking for `user[target_trait]`, and we didn't check `target_trait` value before. That could lead to an error and also is somewhat different behavior that I would expect after reading settings description - in this case target_trait works as a fallback.

File: server/lib/update-user.js [hunk 4]
```diff
@@ -1,141 +1,222 @@
...
+          return skip({
+            reason: "Domain invalid",
+            domain
+          });
+        }
 
-      return datanyze.getDomainInfo(domain).then(data => {
-        asUser.logger.debug("outgoing.user.fetch.response", { response: data });
+        // Skip because we are already enriched
+        const rank = datanyzeAccount.rank || user["traits_datanyze/rank"];
```
Comment (michaloo) [inline server/lib/update-user.js]: I find it a little confusing why we do `user["traits_datanyze/rank"]` but not `account["datanyze/rank"]`.

File: manifest.json (hunk 1 above)
Comment (michaloo) [inline manifest.json]: > If active, Data will be written at Account level in addition to User level

current implementation will write data to either account or user, not both. What should be change here?

Comment (unity) [reply manifest.json]: I don't think we do - At account level, Domain is a recognized field, same as `email` is recognized on the user. On the user we use Email automatically for Clearbit enrich for instance.

File: server/lib/datanyze.js (hunk 2 above)
Comment (unity) [reply server/lib/datanyze.js]: I'm pretty sure they don't - @sbellity can you confirm ?

File: server/lib/update-user.js (hunk 3 above)
Comment (unity) [reply server/lib/update-user.js]: So you'd expect that we simply skip if `handle_account` is true and we don't have an account domain ?

File: server/lib/update-user.js (hunk 4 above)
Comment (unity) [reply server/lib/update-user.js]: `datanyzeAccount.rank` should resolve to  `account.datanyze.rank` (see https://github.com/hull-ships/hull-datanyze/blob/feature/smart-notifier/server/lib/update-user.js#L45)

File: manifest.json (hunk 1 above)
Comment (unity) [reply manifest.json]: We should only write at account level - I'll fix the settings description

File: server/lib/update-user.js (hunk 4 above)
Comment (michaloo) [reply server/lib/update-user.js]: yeah, I know how it works, but I really think we need to stick with one format. More transformation we do in our transactions the more error prone it is. I know you are in favor for grouped format, I don't have anything against it in general, but I'm for using the same stuff we get in notifications and settings dropdowns for mappings. We can obviously change it, but we should be consistent.

File: server/lib/update-user.js (hunk 3 above)
Comment (michaloo) [reply server/lib/update-user.js]: yes, exactly:
handle_accounts === true and there is no account.domain -> skip

**Pull request merged**